autowarefoundation/autoware.universe
# feat(default_ad_api): add door api

## Description

Add default implementation for [door API](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/ad-api/features/vehicle-doors/) and door simulation tools.
- Door API (default_ad_api)
- Door simulator (vehicle_device_simulator)
- Door panel (tier4_adapi_rviz_plugin)

## Related links

https://github.com/autowarefoundation/autoware_launch/pull/921
https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/ad-api/features/vehicle-doors/

## Tests performed

1. Checkout https://github.com/autowarefoundation/autoware_launch/pull/921 and build.
2. Launch planning simulator.
3. Add door panel on rviz (tier4_adapi_rviz_plugin -> DoorPanel)
4. Press open/close button and check status.

![Screenshot from 2023-12-01 13-50-09](https://github.com/autowarefoundation/autoware.universe/assets/43976882/8d6b7541-3ec7-475a-b87d-20b05a890251)

## Notes for reviewers

None

## Interface changes

Add
- `/api/vehicles/doors/command`
- `/api/vehicles/doors/layout`
- `/api/vehicles/doors/status`
- `/vehicles/doors/command`
- `/vehicles/doors/layout`
- `/vehicles/doors/status`

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
